### PR TITLE
Feature/es menu position animation

### LINKED ISSUE #21: How to display it from left instead of right ?
I'm trying to make the context menu coming from left instead of coming from right.

### PR DESCRIPTION
**Pull request by EgorSobko**
This PR resolves #21. It also resolves #22.

### CODE REVIEW

File: YALContextMenu/YALContextMenuTableView.h [hunk 1]
```diff
@@ -8,6 +8,16 @@
 
 #import <UIKit/UIKit.h>
 
+typedef NS_ENUM(NSInteger, MenuItemsSide) {
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.h:11]: :+1:

File: YALContextMenu/YALContextMenuTableView.h [hunk 2]
```diff
@@ -8,6 +8,16 @@
...
+    Left,
+    Right
+};
+
+typedef NS_ENUM(NSInteger, MenuItemsAppearanceDirection) {
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.h:16]: :+1:

File: YALContextMenu/YALContextMenuTableView.h [hunk 3]
```diff
@@ -29,11 +39,16 @@
 
 @interface YALContextMenuTableView : UITableView
 
-@property (nonatomic, weak) id<YALContextMenuTableViewDelegate>yalDelegate;
+@property (nonatomic, weak) id <YALContextMenuTableViewDelegate> yalDelegate;
 
 /*! @abstract animationDuration time for animation in seconds*/
 @property (nonatomic) CGFloat animationDuration;
 
+/*! The side of the screen to which menu items are related(For example, if 'right' chosen then first cell will be opened from right to left and counterclockwise and menu will be closed clockwise). Default - Right.*/
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.h]: @EgorSobko could you please update this text to something like this:
/_! The side of the screen to which menu items are related.
When it set to 'Right' the first cell will be opened from right to left and selected cell will be closed from left to right. 
When it set to 'Left' the first cell will be opened from left to right and selected cell will be closed from right to left. 
Default - Right._/

File: YALContextMenu/YALContextMenuTableView.h [hunk 4]
```diff
@@ -29,11 +39,16 @@
...
+@property (nonatomic) MenuItemsSide menuItemsSide;
+
+/*! @abstract Direction if menu item appearence. Default - FromTopToBottom.*/
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.h]: We need to update this text.

File: YALContextMenu/YALContextMenuTableView.m [hunk 5]
```diff
@@ -186,7 +193,11 @@ - (void)show:(BOOL)show visibleCellsAnimated:(BOOL)animated {
 - (void)dismissBottomCells {
     if (self.bottomCells.count) {
         UITableViewCell<YALContextMenuCell> *hidingCell = [self.bottomCells lastObject];
-        [self show:NO cell:hidingCell animated:YES direction:top clockwise:NO completion:^(BOOL completed) {
+        [self show:NO
+              cell:hidingCell
+          animated:YES
+         direction:self.menuItemsAppearanceDirection == FromBottomToTop ? bottom : top
+         clockwise:self.menuItemsAppearanceDirection == FromBottomToTop ? YES : NO completion:^(BOOL completed) {
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.m]: @EgorSobko please replace the completion handler to the next line

File: YALContextMenu/YALContextMenuTableView.m [hunk 6]
```diff
@@ -199,7 +210,11 @@ - (void)dismissBottomCells {
 - (void)dismissTopCells {
     if (self.topCells.count) {
         UITableViewCell<YALContextMenuCell> *hidingCell = [self.topCells firstObject];
-        [self show:NO cell:hidingCell animated:YES direction:bottom clockwise:YES completion:^(BOOL completed) {
+        [self show:NO
+              cell:hidingCell
+          animated:YES
+         direction:self.menuItemsAppearanceDirection == FromBottomToTop ? top : bottom
+         clockwise:self.menuItemsAppearanceDirection == FromBottomToTop ? NO : YES completion:^(BOOL completed) {
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.m]: please replace the completion handler to the next line

File: YALContextMenu/YALContextMenuTableView.m [hunk 7]
```diff
@@ -326,11 +356,13 @@ - (id)dequeueReusableCellWithIdentifier:(NSString *)identifier forIndexPath:(NSI
...
         if (self.animatingState==Showing) {
             [cell contentView].hidden = YES;
-        }else if (self.animatingState==Stable)
-        {
+        } else if (self.animatingState==Stable) {
             [cell contentView].hidden = NO;
             [cell animatedContent].alpha = 1;
         }
+        if (self.menuItemsAppearanceDirection == FromBottomToTop) {
+            cell.layer.transform = CATransform3DRotate(CATransform3DIdentity, M_PI, 0.0f, 0.0f, 1.0f);
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.m:366]: :+1:

File: YALContextMenu/YALContextMenuTableView.m [hunk 8]
```diff
@@ -364,4 +396,13 @@ - (void)setAnchorPoint:(CGPoint)anchorPoint forView:(UIView *)view {
     view.layer.anchorPoint = anchorPoint;
 }
 
+#pragma mark - Setters
+- (void)setMenuItemsAppearanceDirection:(MenuItemsAppearanceDirection)menuItemsAppearanceDirection {
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.m:402]: :+1:

File: YALContextMenu/YALContextMenuTableView.m [hunk 9]
```diff
@@ -364,4 +396,13 @@ - (void)setAnchorPoint:(CGPoint)anchorPoint forView:(UIView *)view {
...
+    if (menuItemsAppearanceDirection != _menuItemsAppearanceDirection) {
+        _menuItemsAppearanceDirection = menuItemsAppearanceDirection;
+        if (self.menuItemsAppearanceDirection == FromBottomToTop) {
+            self.layer.transform = CATransform3DRotate(CATransform3DIdentity, M_PI, 0.0f, 0.0f, 1.0f);
```
Comment (maksym-letiushov) [inline YALContextMenu/YALContextMenuTableView.m:406]: An elegant solution :+1: Thanks

**Comment by maksym-letiushov** [comment]
@EgorSobko please make just a minor fixes and good to merge. Thanks.

File: YALContextMenu/YALContextMenuTableView.h (hunk 3 above)
Comment (EgorSobko) [inline YALContextMenu/YALContextMenuTableView.h]: Done.

File: YALContextMenu/YALContextMenuTableView.m (hunk 5 above)
Comment (EgorSobko) [inline YALContextMenu/YALContextMenuTableView.m]: Done.

File: YALContextMenu/YALContextMenuTableView.m (hunk 6 above)
Comment (EgorSobko) [inline YALContextMenu/YALContextMenuTableView.m]: Done.

**Comment by maksym-letiushov** [comment]
thanks

**Pull request merged**